### DISCUSSION TITLE
【分享】使用开源的 DAPLink 调试器及 pyOCD 烧录程序

### DISCUSSION THREAD

**Discussion by xingrz** [peng-zhihui/HelloWord-Keyboard#76] [Show and tell]
README 里提到，我们可以通过 SWD 口用 JLink 或者 STLink 烧录。但是为了将开源贯彻到底，我们其实还可以用开源的 DAPLink 调试器和 pyOCD 烧录，支持 Windows/Linux/Mac。这里大致跟大家分享一下怎么做。

### 准备工具

1. 一个 DAPLink 调试器…某宝一搜一大把几十块钱的，甚至随便找一个 STM32F103CBT6 的调试器也行，甚至…用瀚文刷瀚文？
2. [pyOCD](https://pyocd.io/docs/installing)
3. 对应的 CMSIS 支持包…打开[这个网址](https://www.keil.com/pack)，浏览器搜索 STM32F1 找到这一条展开，点击 Download 下载，得到一个文件 `Keil.STM32F1xx_DFP.2.4.0.pack`
    <img width="706" alt="image" src="https://user-images.githubusercontent.com/288288/185315763-e09b7df0-ab48-493c-92fd-9fad5a791c06.png">

### 如何连接

准备三根杜邦线，将 GND/SWDIO/SWCLK 三个引脚接到 DAPLink 对应引脚上。

### 如何烧录

执行命令：

```sh
 pyocd load --pack Keil.STM32F1xx_DFP.2.4.0.pack --target stm32f103cb -a 0x08000000 HelloWord-Keyboard-fw.bin
```

完成！

下一篇将会跟大家分享一个 Bootloader，不借助调试器直接通过 USB 更新固件。

**Comment by CowardTieMeng** [comment]
大佬，刷完固件再插上键盘，windows提示无法识别设备
![QQ图片20221225221738](https://user-images.githubusercontent.com/44440346/209471478-599f1258-c0b7-40c5-a0b2-d93079521478.png)

**Reply by CowardTieMeng** [reply]
解决了，在智晖君的包里找到了USB驱动安装工具，安装完就能识别了

**Reply by xujianle** [reply]
请教一下我的情况和你一样，应该怎么操作呢？zadig我选择了未知设备安装libusb可是安装失败

**Reply by CowardTieMeng** [reply]
去智辉君的git项目那里，下载下来，里面有个usb driver，插上线等那个装完就好了

**Reply by xujianle** [reply]
> 去智辉君的git项目那里，下载下来，里面有个usb driver，插上线等那个装完就好了

是瀚文项目里tool文件夹里面zadig吗？

**Reply by CowardTieMeng** [reply]
最好截图看下，我安装的时候挺顺利的

**Reply by xujianle** [reply]
> 最好截图看下，我安装的时候挺顺利的

能加您个联系方式吗q 1820200950
